### PR TITLE
fix: [CDS-77857]: Removing usage of ModelMappers from pipeline filters.

### PR DESCRIPTION
**Pull request by Aninditaa-ch**
## Describe your changes
**Issue:** Filters API is intermittently throwing error 500 after creation of a custom filter. 

**Changes:** In PipelineExecutionFilterPropertiesMapper, PipelineFilterPropertiesMapper and TemplateFilterPropertiesMapper, we have written the Mappers using writeDto and toEntity methods for each of their Filter Properties.

**Testing:** Added the java test files for the three Mappers classes created.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)

### CODE REVIEW

**Review by shalini-agr — COMMENTED** [review]
> Reviewed 2 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java` line 50 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeM73G86_h5ueezCpuS:-NeM73G86_h5ueezCpuT:b730wp3) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java#L50)):*
> ```Java
>             .description(properties.getDescription())
>             .templateEntityTypes(properties.getTemplateEntityTypes())
>             .description(properties.getDescription())
> ```

description has been set twice, remove this one.

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 17 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeM7SzZELaLkKa34zBw:-NeM7SzZELaLkKa34zBx:bv2epsz) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L17)):*
> ```Java
> import io.harness.pms.pipeline.PipelineFilterPropertiesDto;
> 
> public class PipelineFilterPropertiesMapper
> ```

Add unit test for this class
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java` line 18 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeM7Uh3Du_qo4xCq9KU:-NeM7Uh3Du_qo4xCq9KV:b4ujuh7) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java#L18)):*
> ```Java
> 
> public class PipelineExecutionFilterPropertiesMapper
>     implements FilterPropertiesMapper<PipelineExecutionFilterPropertiesDTO, PipelineExecutionFilterProperties> {
> ```

Add unit test for this

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        assertEquals(pipelineFilterProperties.getPipelineTags(),pipelineFilterPropertiesDto.getPipelineTags());
+        assertEquals(pipelineFilterProperties.getName(),pipelineFilterPropertiesDto.getName());
+        assertEquals(pipelineFilterProperties.getModuleProperties(),pipelineFilterPropertiesDto.getModuleProperties());
+        assertEquals(pipelineFilterProperties.getDescription(),pipelineFilterPropertiesDto.getDescription());
+    }
+
+    @Test
+    @Owner(developers = ANINDITAA)
+    @Category(UnitTests.class)
+    public void testPipelineEntity() {
```
Comment (shalini-agr) [inline pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java:63]: change test name to testPipelineFilterWriteDTO as you are testing writeDTO method here

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    @Owner(developers = ANINDITAA)
+    @Category(UnitTests.class)
+    public void testPipelineWriteDTO() {
```
Comment (shalini-agr) [inline pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java:43]: change test name to testPipelineFilterToEntity as you are testing toEntity method here

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        assertEquals(pipelineExecutionFilterProperties.getPipelineTags(),pipelineExecutionFilterPropertiesDTO.getPipelineTags());
+        assertEquals(pipelineExecutionFilterProperties.getModuleProperties(),pipelineExecutionFilterPropertiesDTO.getModuleProperties());
+        assertEquals(pipelineExecutionFilterProperties.getStatus(),pipelineExecutionFilterPropertiesDTO.getStatus());
+        assertEquals(pipelineExecutionFilterProperties.getTags(),pipelineExecutionFilterPropertiesDTO.getTags());
+    }
+
+    @Test
+    @Owner(developers = ANINDITAA)
+    @Category(UnitTests.class)
+    public void testPipelineFilterEntity() {
```
Comment (shalini-agr) [inline pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java:66]: change test name to testPipelineExecutionFilterWriteDTO as you are testing writeDTO in this method

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    @Owner(developers = ANINDITAA)
+    @Category(UnitTests.class)
+    public void testPipelineFilterWriteDTO() {
```
Comment (shalini-agr) [inline pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java:46]: change test name to testPipelineExecutionFilterToEntity as you are testing toEntity method here

**Review by shalini-agr — CHANGES_REQUESTED** [review]

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java (hunk 1 above)
Comment (Aninditaa-ch) [reply pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java]: Completed.

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java (hunk 2 above)
Comment (Aninditaa-ch) [reply pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineFilterPropertiesMapperTest.java]: Completed.

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java (hunk 3 above)
Comment (Aninditaa-ch) [reply pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java]: Completed.

File: pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java (hunk 4 above)
Comment (Aninditaa-ch) [reply pipeline-service/service/src/test/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapperTest.java]: Completed.

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by Aninditaa-ch — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 7 file(s) to review. 1 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 17 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeM7SzZELaLkKa34zBw:-NfKyeIwBH38kSDKu9o1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L17)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

Add unit test for this class
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java` line 18 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeM7Uh3Du_qo4xCq9KU:-NfKydlXEJn_EXadH821:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java#L18)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

Add unit test for this
</blockquote></details>

Done.
___
*[`template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java` line 50 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeM73G86_h5ueezCpuS:-NfKy_m08V53I1fSPT9Y:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java#L50)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

description has been set twice, remove this one.
</blockquote></details>

Done.

**Review by shalini-agr — APPROVED** [review]
> Reviewed 1 of 2 files at r4, 2 of 3 files at r5, 2 of 3 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 7 file(s) to review. 4. unresolved discussions. 1 of 2 approvals obtained.

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 45 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeIC3De-4JSvhbvQCAW:-NeIC3De-4JSvhbvQCAX:bw7c7to) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L45)):*
> ```Java
>             .build();
> 
>     pipelineFilterProperties.setType(pipelineFilterPropertiesDto.getFilterType());
> ```

We need not explicitly set this, we can directly add it similar to other properties like pipelineTags

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL6vAmCQ-WUYR7AieD:-NfL6vAmCQ-WUYR7AieE:bnj7hl0) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java#L31)):*
> ```Java
>             .timeRange(executionFilterProperties.getTimeRange())
>             .build();
>     pipelineExecutionFilterPropertiesDTO.setTags(TagMapper.convertToMap(pipelineExecutionFilterProperties.getTags()));
> ```

same as above

**Comment by sahilhindwani-harness** [comment]
*[`template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java` line 51 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL730G0Ob0NdnML04C:-NfL730G0Ob0NdnML04D:b-byc8id) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java#L51)):*
> ```Java
>             .repoName(properties.getRepoName())
>             .build();
>     templateFilterPropertiesDTO.setFilterType(filterProperties.getType());
> ```

nit: these can be added directly in builder itself.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 6 file(s) to review. 1 change requests. 4. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfL7Qb-Exlo8GrL4o_v:b-hu9bq1) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
> ```Java
>             .build();
> 
>     pipelineFilterPropertiesDto.setTags(TagMapper.convertToMap(pipelineFilterProperties.getTags()));
> ```

why we are setting separately? and not in builder?

**Comment by Aninditaa-ch** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfMKymBDL-bGXLSqQQw:b-ix2tn8) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why we are setting separately? and not in builder?
</blockquote></details>

To set it in the same, we have to add Superbuilder annotation for all the subsequent classes, not working without that.
cc: @sahilhindwani-harness @shalini-agr

**Review by archit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 6 file(s) to review. 1 change requests. 6. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfPCrwhD8iXFP7aljV_:byurxx4) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, Aninditaa-ch wrote…</i></summary><blockquote>

To set it in the same, we have to add Superbuilder annotation for all the subsequent classes, not working without that.
cc: @sahilhindwani-harness @shalini-agr
</blockquote></details>

is there any reason of not using it then?

There are other way to extend builder also but just checking if we know why superbuilder 

Ref- https://www.baeldung.com/lombok-builder-inheritance

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 6 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfPPunV3a6qufpk-Vpx:bexpcpe) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

is there any reason of not using it then?

There are other way to extend builder also but just checking if we know why superbuilder 

Ref- https://www.baeldung.com/lombok-builder-inheritance
</blockquote></details>

In the case of inheritance, SuperBuilder is simpler to use, the only problem is it needs to be added to all the children's classes. So, if we add that annotation to FilterProperties and FilterPropertiesDTO, we'll need to change all the DTOs extending them. and there are like 15 such DTOs in different services.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 7 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfQlzHB6pbTI-neAsPI:b-w9nzor) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

In the case of inheritance, SuperBuilder is simpler to use, the only problem is it needs to be added to all the children's classes. So, if we add that annotation to FilterProperties and FilterPropertiesDTO, we'll need to change all the DTOs extending them. and there are like 15 such DTOs in different services.
</blockquote></details>

@shalini-agr what if we use the other approach to include constructor?

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfnjN87Dvuc2UMILj-b:b-gubeaf) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@shalini-agr what if we use the other approach to include constructor?
</blockquote></details>

I also think we should set it in the same place. either via constructor or using SuperBuilder or creating a custom builder.

**Comment by shalini-agr** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfnrXvdEtGUCR6wzzb9:b-r4jtdj) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

I also think we should set it in the same place. either via constructor or using SuperBuilder or creating a custom builder.
</blockquote></details>

yes, discussed it with Aninditaa, she will create custom builders in our filters and DTOs to be able to set all the fields in the builder itself.

**Review by Aninditaa-ch — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 15 file(s) to review. 1 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 45 at r1](https://reviewable.io/reviews/harness/harness-core/52916#-NeIC3De-4JSvhbvQCAW:-NfoS8bp1S_vjNo8l_UG:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9f215319963718017bad070d86563462be440dd0/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L45)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

We need not explicitly set this, we can directly add it similar to other properties like pipelineTags
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL7Qb-Exlo8GrL4o_u:-NfoSFNN9KLz-jrEmQ1_:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/pipeline/mappers/PipelineFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

yes, discussed it with Aninditaa, she will create custom builders in our filters and DTOs to be able to set all the fields in the builder itself.
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java` line 31 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL6vAmCQ-WUYR7AieD:-NfoSHjpA3tvKb321Pri:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/mapper/PipelineExecutionFilterPropertiesMapper.java#L31)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Done.
___
*[`template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java` line 51 at r8](https://reviewable.io/reviews/harness/harness-core/52916#-NfL730G0Ob0NdnML04C:-NfoSLb082Hwj19gRy4l:b-rnrbll) ([raw file](https://github.com/harness/harness-core/blob/5dc32f65a75a1f1a79ff351dbff82ee197a11626/template-service/service/src/main/java/io/harness/template/mappers/TemplateFilterPropertiesMapper.java#L51)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

nit: these can be added directly in builder itself.
</blockquote></details>

done

**Review by archit-harness — DISMISSED** [review]
> Reviewed 1 of 3 files at r9, 13 of 14 files at r10, 1 of 1 files at r11, 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 15 file(s) to review. 6. unresolved discussions. 1 of 2 approvals obtained.

**Comment by shalini-agr** [comment]
trigger uts

**Review by shalini-agr — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 15 file(s) to review. 6. unresolved discussions.

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 15 file(s) to review. 6. unresolved discussions. 1 of 2 approvals obtained.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r4, 11 of 14 files at r10, 1 of 1 files at r11, 1 of 1 files at r12, 2 of 2 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52916)* status: 4. unresolved discussions.

**Comment by Aninditaa-ch** [comment]
trigger review

**Pull request merged**